python/ray/autoscaler/commands.py

```diff
@@ -630,10 +664,10 @@ def get_or_create_head_node(config,
                 init_commands = []
                 ray_start_commands = config["head_start_ray_commands"]
             elif no_restart:
-                init_commands = config["head_setup_commands"]
+                init_commands = _get_head_setup_commands(config)
```

Comment:
As discussed before, can't we simply keep this as `head_setup_commands` (just ignore node specific setup)?

Reply:
wait i thought we discussed node specific setup commands overriding both `head_setup_commands` and `setup_commands` (at the config level)

Reply:
Only for worker node, not for the head. The head node is never configured by per-node-type commands, which simplifies things a lot. This isn't a problem for the user since they can always put whatever head setup they want in the head setup command section.